ocaml/xapi-storage-script/main.ml

```diff
@@ -349,6 +353,27 @@ module Script = struct
         update_mapping ~script_dir >>= fun () -> find ()
 end
 
+let observer_config_dir =
+  let ( // ) = Filename.concat in
+  let dir, component =
+    Constants.(observer_config_dir, observer_component_smapi)
+  in
+  dir // component // "enabled"
+
+(** Determine if SM API observation is enabled from the
+    filesystem. Ordinarily, determining if a component is enabled
+    would consist of querying the 'components' field of an observer
+    from the xapi database. *)
+let observer_is_component_enabled () =
+  let ( let* ) = ( >>= ) in
+  let is_enabled () =
+    let is_config_file path = Filename.check_suffix path ".observer.conf" in
+    let* files = Sys.readdir observer_config_dir in
```

Comment:
May I please know how does the `let*` work here?

Reply:
Thanks for the elaboration!
I missed that the `Sys.readdir` is actually from https://github.com/janestreet/async_unix/blob/master/src/async_unix.ml#L19
I thought it's native `Sys.readdir` wrongly, so I couldn't understand the bind operator there.